src/component/variantPage/BasicInfo.tsx

```diff
@@ -1,127 +1,439 @@
 import * as React from 'react';
-import './BasicInfo.css';
-import { observer } from 'mobx-react';
-import { Row, Col } from 'react-bootstrap';
-import { VariantAnnotationSummary } from 'cbioportal-frontend-commons';
 import _ from 'lodash';
+import { observer } from 'mobx-react';
+import classNames from 'classnames';
+import autobind from 'autobind-decorator';
+import { observable, action } from 'mobx';
+import {
+    VariantAnnotationSummary,
+    TranscriptConsequenceSummary,
+    getCanonicalMutationType,
+    DefaultTooltip,
+} from 'cbioportal-frontend-commons';
+import { Alteration } from 'cbioportal-frontend-commons/api/generated/OncoKbAPI';
+import { Mutation } from 'react-mutation-mapper';
+import basicInfo from './BasicInfo.module.scss';
+import { Button } from 'react-bootstrap';
+import TranscriptSummaryTable from './TranscriptSummaryTable';
 
 interface IBasicInfoProps {
     annotation: VariantAnnotationSummary | undefined;
+    mutation: Mutation;
+    oncokbVariant?: Alteration[];
+    variant: string;
 }
 
+type MutationTypeFormat = {
+    label?: string;
+    className: string;
+};
+
+type OncogeneTsg = {
+    oncogene?: boolean | undefined;
+    tsg?: boolean | undefined;
+};
+
+export const MAIN_MUTATION_TYPE_MAP: { [key: string]: MutationTypeFormat } = {
```

Comment:
we should probably get this from `cbioportal-frontend-commons`, maybe add a TODO for this?

Reply:
@onursumer I copied it from react-mutation-mapper, cause I can't import it directly from there. I add a TODO to mention probably we can get it from both